themes/bootstrap3/templates/Helpers/star-rating.phtml

```diff
@@ -0,0 +1,31 @@
+<?php
+  $id = $this->escapeHtmlAttr(uniqid());
+  $labelAttrs = !empty($this->labelAttrs) ? (' ' . implode(' ', $this->labelAttrs)) : '';
+  $inputAttrs = !empty($this->inputAttrs) ? (' ' . implode(' ', $this->inputAttrs)) : '';
+  $rating = $ratingData['count'] ? ($ratingData['rating'] ?? 0) : null;
+  // Unique name for the fields of read-only ratings so that displaying multiple ratings on a page works:
+  $fieldName = $this->readonly ? "rating_{$id}" : 'rating';
+?>
+<div class="star-rating<?=$this->readonly ? ' readonly' : ''?>">
+  <label class="rating__label hidden" for="rating_<?=$id?>_none" aria-label="<?=$this->transEsc('rating_none')?>"<?=$labelAttrs?>>
+    &nbsp;
+  </label>
+  <input class="rating__input hidden" name="<?=$fieldName?>" id="rating_<?=$id?>_none" value="" type="radio" autocomplete="off"<?=null === $rating ? 'checked' :''?><?=$inputAttrs?>>
+  <label class="rating__label" for="rating_<?=$id?>_0" aria-label="<?=$this->transEsc('rating_0')?>"<?=$labelAttrs?>>
+    &nbsp;
+  </label>
+  <input class="rating__input" name="<?=$fieldName?>" id="rating_<?=$id?>_0" value="0" type="radio" autocomplete="off"<?=0 === $rating ? 'checked' :''?><?=$inputAttrs?>>
+  <?php for ($level = 10; $level <= 100; $level += 10): ?>
+    <?php
+      $fieldId = "rating_{$id}_$level";
+      $rangeLow = $level;
+      $rangeHigh = 100 === $level ? 101 : ($level + 10);
+      $class = $level % 20 === 10 ? ' rating__label--half' : '';
+      $icon = $this->icon($level % 20 === 10 ? 'rating-half' : 'rating-full', 'rating__icon rating__icon--star');
+    ?>
+    <label class="rating__label<?=$class?>" for="<?=$fieldId?>" aria-label="<?=$this->transEsc("rating_$level")?>"<?=$labelAttrs?>>
+      <?=$icon?>
+    </label>
+    <input class="rating__input" name="<?=$fieldName?>" id="<?=$fieldId?>" value="<?=$level ?>" type="radio" autocomplete="off"<?=$rating >= $rangeLow && $rating < $rangeHigh ? ' checked' : ''?><?=$this->readonly ? ' disabled' : ''?><?=$inputAttrs?>>
```

Comment:
1. Could we abstract the label+input pair into a local function to prevent errors?
2. We can add `autocomplete=off` to the containing form if that helps simplify the HTML.

Reply:
Not sure I understand what this would prevent. Can you clarify?

Reply:
1. At the time, I thought it would simplify the code and prevent future errors modifying this code.
2. Just minimizing html.

Reply:
@crhallberg Sorry, I had completely lost track of this. 

1. I think the elements are suffiently different to warrant the two different cases.
2. Oops, looks like `autocomplete=off` isn't even supported for radio buttons. Now removed.